### PR TITLE
Sign-up

### PR DESCRIPTION
**Pull request by Margo1212**

### CODE REVIEW

File: src/utilities/Button.tsx [hunk 1]
```diff
@@ -2,14 +2,16 @@ import { ButtonHTMLAttributes } from "react";
 
 type Props = {
```
Comment (aryla12) [inline src/utilities/Button.tsx]: For Components like button you extends you Component Props with ButtonHTMLAttributes from react library.

File: src/utilities/Input.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import { ErrorMessage, Field } from "formik";
+
+type Props = {
```
Comment (aryla12) [inline src/utilities/Input.tsx]: You can use/extend InputHTMLAttributes<HTMLInputElement>.

File: src/app/(auth)/sign-up/signUpValidationSchema.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import { z } from "zod";
+
+export const SignUpValidationSchema = z
+  .object({
```
Comment (aryla12) [inline src/app/(auth)/sign-up/signUpValidationSchema.ts:4]: Should any of that be required?

File: src/app/(auth)/sign-up/SignUpForm.tsx [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        return error.formErrors.fieldErrors;
+      }
+    }
+  };
+
+  const handleSubmit = async (values: SignUpFormValues) => {
+    try{
+      console.log(values);
+    } catch (error) {
+      console.error('error', error)
```
Comment (aryla12) [inline src/app/(auth)/sign-up/SignUpForm.tsx]: Try to use as less console.log as you can. Errors should be displayed in the view. If something is in progress like the values above add comment with `TODO:` and some info about it

File: src/app/(auth)/sign-up/SignUpForm.tsx [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+  return (
+        <form onSubmit={handleSubmit(onSubmit)} className="py-10 space-y-5 w-full h-full tablet:h-auto flex flex-col">
+          <Input
+            type="text"
+            label="Nazwa użytkownika"
+            placeholder="Nazwa użytkownika"
+            id="username"
+            {...register('username')}
+                error={errors.username}
```
Comment (aryla12) [inline src/app/(auth)/sign-up/SignUpForm.tsx]: too much indentation here?

File: src/app/(auth)/sign-up/SignUpForm.tsx [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+          />        
+          <Input
+            type="email"
+            label="Adres Email"
+            placeholder="example@example.com"
+            id="email"
+            {...register('email')}
+                error={errors.email}
```
Comment (aryla12) [inline src/app/(auth)/sign-up/SignUpForm.tsx]: too much indentation here?

File: src/app/(auth)/sign-up/SignUpForm.tsx [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+          />       
+          <Input
+            type="password"
+            label="Hasło"
+            placeholder="********"
+            id="password"
+            {...register('password')}
+                error={errors.password}
```
Comment (aryla12) [inline src/app/(auth)/sign-up/SignUpForm.tsx]: too much indentation here?

File: src/app/(auth)/sign-up/SignUpForm.tsx [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+          />        
+          <Input
+            type="password"
+            label="Powtórz Hasło"
+            placeholder="********"
+            id="password2"
+            {...register('password2')}
+                error={errors.password2}
```
Comment (aryla12) [inline src/app/(auth)/sign-up/SignUpForm.tsx]: too much indentation here?

**Review by aryla12 — CHANGES_REQUESTED** [review]

File: src/app/(auth)/sign-up/signUpValidationSchema.ts (hunk 3 above)
Comment (Margo1212) [reply src/app/(auth)/sign-up/signUpValidationSchema.ts:4]: Thank you for your suggestion!  In Zod, fields are required by default unless explicitly marked as optional. (z.optional() or z.nullable()).

**Review by aryla12 — APPROVED** [review]

**Pull request merged**